### DISCUSSION TITLE
Companion Mode with G7 Getting out of date and double readings

### DISCUSSION THREAD

**Discussion by grahamjenson** [NightscoutFoundation/xDrip#2803] [General]
I am wondering if anyone else is seeing the issues with G7 in companion mode where sometimes it misses readings or gives the same reading twice. And it seems to get out of sync and get behind by minutes.

I used xDrip in companion mode with CamAPS before and it did not have these issues. I am thinking it might be an issue with the G7 App sending multiple notifications, or something. I am a bit familiar with the UIBasedCollector code, if there is anything I can do to help please let me know.

**Comment by Arakon** [comment]
I've had no such issue on a Pixel 6a. However, when backfilling after a long signal loss, only 1 value every 40 minutes seems to be added, while the actual G7 app gets the full backfill. 
Edit to clarify: This happens when backfilling several hours, if the backfill is only 30 mins or so, all data is there.

Signal loss at 3 am until 6 am:

![Screenshot_20230428-063412](https://user-images.githubusercontent.com/12016360/235055109-e3d7646f-3610-4eec-8bca-c9edcf4143f0.png)
